connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -226,6 +226,7 @@ public void stop() {
 
         offsetBackingStore.stop();
         metrics.stop();
+        stopExecutor();
```

Comment:
Are there advantages of putting this simple if-check in a separate methods? Would it be simpler and more straightforward to just do the check here:
```suggestion
        if (executor != null) {
            executor.shutdownNow();
        }
```
and then remove the `stopExecutor()` method?

There is already precedence for an if-check a few lines above.